### DISCUSSION TITLE
storybook with angular and features like input / model signals, css @import and so on

### DISCUSSION THREAD

**Discussion by maku** [storybookjs/storybook#27519] [Help]
### Summary

Is there an example of using storybook with the latest features of angular like input / model signals,  css @import and so on. I tried storybook with a new angular project but I struggled with this topics, so I gave up. Now I want to give it another try. But I do not find resources / examples ho to do it... 

Can anybody give me a hint...

**Comment by Marklb** [comment]
I don't think anything should be different with how stories are defined for signal inputs/outputs, since they are just `args` to Storybook. A model input is recognized as an input, but I don't think there is really a way to utilize it's two-way binding.

I am not sure what you are referring to with a change that would involve css `@import`, but Storybook let's Angular's builder handle styles. So, there shouldn't be anything Specific to Storybook about that, that I know of.

As for something to look at, you could maybe look at this code from the development sandbox, but there isn't anything special in them that I can tell. https://github.com/storybookjs/storybook/tree/03ef5c6af4a47d197fe130b53216fd1f25010ede/code/frameworks/angular/template/stories_angular-cli-default-ts/signal

**Comment by maku** [comment]
Ok, so now more concrete. e.g.

**css**

when I import the angular css file in .storybook/preview.ts I get errors:

ERROR in ./src/styles.css?ngGlobalStyle 1:0
Module parse failed: Unexpected character '@' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> @import '@angular/cdk/overlay-prebuilt.css';
| @import "styles/variables.css";
| @import "styles/flex.css";
 @ ./.storybook/preview.ts 1:0-27

I would think that, when using simple plain default angular project structure, I do not have to define a loader. I found nothing in documentation.


**signal inputs**

It seems this works better now...  


But in general it is really hard to get info / documentation etc. for storybook / angular stuff

**Reply by valentinpalkovic** [reply]
Styles can be cofigured like you would configure them for your Angular application via your angular.json workspace file. Just define your styles as an option on the Storybook Builder. All available builder options are documented here: https://storybook.js.org/docs/8.0/get-started/angular#how-do-i-configure-angulars-builder-for-storybook

Input signals are partially supported. Storybook's controls are currently not able to reliably extract the type definitions for input signals due to a bug in Compodoc, which we use to analyze a components signature and input properties. Currently, you have to use argTypes to define types explicitly for input signals, as described here: https://storybook.js.org/docs/api/arg-types
